presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -817,6 +817,9 @@ private PhysicalOperation visitScanFilterAndProject(
                     return new PhysicalOperation(operatorFactory, outputMappings, source);
                 }
             }
+            catch (PrestoException pe) {
```

Comment:
This handles the new validation I added to the constructor of the ``JsonPathTokenizer`` above. It can throw a ``PrestoException`` and this exception was getting wrapped into another ``PrestoException[COMPILER_ERROR]`` due to the catch block that catches the ``RuntimeExceptions`` below.